app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt

```diff
@@ -736,6 +736,11 @@ class BrowserFragment : Fragment(), BackHandler, CoroutineScope {
 
                 super.onLoadingStateChanged(session, loading)
             }
+
+            override fun onTitleChanged(session: Session, title: String) {
```

Comment:
I think `onUrlChanged` is better here. In case the title never changes.

Reply:
Yep was trying to avoid the most calls to search bookmarks but if we have to for accuracy we have to

Reply:
> Yep was trying to avoid the most calls to search bookmarks but if we have to for accuracy we have to

Excellent point! You probably want to cancel the previous coroutine first before launching the next one.